RedisRepo/RedisContext.cs

```diff
@@ -265,10 +266,18 @@ public static TResult Retry<TResult>(Func<TResult> func, int maxAttempts)
         {
             return RetryUtil.Retry(func, maxAttempts);
         }
+        public async static Task<TResult> RetryAsync<TResult>(Func<Task<TResult>> func, int maxAttempts)

```

Comment:
In cases where you have methods that can only return the task without ``` await```-ing it, please do. in this case, the method should look something like this:
``` csharp
public static Task<TResult> RetryAsync<TResult>(Func<Task<TResult>> func, int maxAttempts)
{
    return RetryUtil.RetryAsync(func, maxAttempts);
}
``` 

the responsibility to await moves to the caller (keep in mind that the caller would have awaited the result anyway) this way you have one less async-await state machine in your result